test/test_type_promotion.py

```diff
@@ -808,6 +808,48 @@ def test_integer_addcdiv_deprecated(self, device, dtype):
         with self.assertRaisesRegex(RuntimeError, '^Integer division.+is no longer supported+'):
             t.addcdiv_(t, t)
 
+    def test_addcdiv_promotion(self, device):
+        types = (
+            (torch.float64, torch.float64, torch.complex128),
+            (torch.long, torch.bfloat16, torch.float32),
+        )
+
+        for type1, type2, type3 in types:
+            # Cannot use rand with long.
+            if type1 == torch.long:
+                arg1 = torch.randint(7, [5, 5], device=device, dtype=type1)
+            else:
+                arg1 = torch.rand([5, 5], device=device, dtype=type1)
+            arg2 = torch.rand([5, 5], device=device, dtype=type2)
+            arg3 = torch.rand([1, 5], device=device, dtype=type3)
+
+            res1 = torch.addcdiv(arg1, arg2, arg3)
```

Comment:
`res1` and `res2` are actually not guaranteed to be the same because our type promotion is kinda whacky and not associative. They are the same when the all the inputs are tensors with one or more dimensions

Reply:
thx for the info! added a comment, doesn't apply as long as the shapes are hardcoded (as they are now)